pebblo/app/service/local_ui/loader_apps.py

```diff
@@ -0,0 +1,490 @@
+    def _get_top_n_findings(raw_data):
+        """
+        Return top N findings from all findings
+        """
+        logger.debug("Getting top N findings details and aggregate them")
+        documents_with_findings = raw_data["documentsWithFindings"]
+        logger.debug(f"Doc: {len(documents_with_findings)}")
+        top_n_findings_list = documents_with_findings[
+            : ReportConstants.TOP_FINDINGS_LIMIT.value
+        ]
+        logger.debug(f"NFindigns: {top_n_findings_list}")
+        logger.debug(len(top_n_findings_list))
+        top_n_findings = []
+        for findings in top_n_findings_list:
+            finding_obj = {
+                "fileName": findings["sourceFilePath"],
+                "fileOwner": "-"
+                if findings.get("fileOwner", "-") is None
+                else findings.get("fileOwner", "-"),
+                "sourceSize": 0
+                if findings.get("sourceSize", 0) is None
+                else findings.get("sourceSize", 0),
+                "findingsEntities": findings["findingsEntities"],
+                "findingsTopics": findings["findingsTopics"],
+                "findings": findings["findingsEntities"] + findings["findingsTopics"],
+                "authorizedIdentities": findings.get("authorized_identities", []),
+            }
+            top_n_findings.append(finding_obj)
+        return top_n_findings
+
+    @staticmethod
+    def _create_data_source_findings_summary(data_source_findings):
+        """
+        Creating data source findings summary and return it findings summary list
+        """
+        logger.debug("Creating data source summary")
+        data_source_findings_summary = []
+        for ds_findings in data_source_findings:
+            label_name = ds_findings.get("labelName", "")
+            findings = ds_findings.get("findings", 0)
+            findings_type = ds_findings.get("findingsType")
+            snippet_count = ds_findings.get("snippetCount", 0)
+            file_count = ds_findings.get("fileCount", 0)
+
+            data_source_findings_summary.append(
+                {
+                    "labelName": label_name,
+                    "findings": findings,
+                    "findingsType": findings_type,
+                    "snippetCount": snippet_count,
+                    "fileCount": file_count,
+                }
+            )
+
+        return data_source_findings_summary
+
+    def _get_data_source_details(self, app_data, raw_data):
+        """
+        Create data source findings details and data source findings summary
+        """
+        logger.debug("Aggregating data source details")
+        data_source_obj_list = []
+        for loader in app_data["loaders"]:
+            name = loader.get("name")
+            source_path = loader.get("sourcePath")
+            source_type = loader.get("sourceType")
+            source_size = loader.get("sourceSize")
+            total_snippet_count = raw_data["dataSource"][0][
```

Comment:
I think we should check if raw_data["dataSource"] its size is greater than 0

Reply:
Updated.